File: test/new_lang/test8/TestProgram.pas
program TestProgram;
uses SplashKit;

function AFunc(x: Integer): Integer; cdecl;
begin
  WriteLn('Hello', x);
  result := x + 10;
end;

procedure AProc(x: Integer); cdecl;
begin
  WriteLn('Hello World', x);
end;

procedure Main();
begin
  RunFunc(@AFunc);
  RunProc(@AProc);
end;

begin
  Main();
end.

File: test/new_lang/test8/TestProgram.cs
using static SplashKitSDK.SplashKit;
using static System.Console;
using System.Collections.Generic;

public class Program
{
  private static int AFunc(int x)
  {
    WriteLine("Hello {0}", x);
    return x + 10;
  }

  private static void AProc(int x)
  {
    WriteLine("Hello World {0}", x);
  }

  public static void Main()
  {
    RunFunc(AFunc);
    RunProc(AProc);
  }
}
